test/common/buffer/owned_impl_test.cc

```diff
@@ -794,188 +783,155 @@ TEST_F(OwnedImplTest, ReserveCommit) {
-    expectSlices({{1, 4095, 4096}, {0, 4096, 4096}, {0, 4096, 4096}}, buffer);
-    EXPECT_EQ(3, num_reserved);
-    EXPECT_EQ(slice1, iovecs[0].mem_);
-    EXPECT_EQ(slice2, iovecs[1].mem_);
-    clearReservation(iovecs, num_reserved, buffer);
+    {
+      expectSlices({{1, 16383, 16384}}, buffer);
+      auto reservation = buffer.reserveApproximately(32768);
+      EXPECT_EQ(2, reservation.numSlices());
+      EXPECT_EQ(16383, reservation.slices()[0].len_);
+      EXPECT_EQ(16384, reservation.slices()[1].len_);
+    }
 
     // Append a fragment to the buffer, and then request a small reservation. The buffer
     // should make a new slice to satisfy the reservation; it cannot safely use any of
     // the previously seen slices, because they are no longer at the end of the buffer.
-    expectSlices({{1, 4095, 4096}}, buffer);
-    buffer.addBufferFragment(fragment);
-    EXPECT_EQ(13, buffer.length());
-    num_reserved = buffer.reserve(1, iovecs, NumIovecs);
-    expectSlices({{1, 4095, 4096}, {12, 0, 12}, {0, 4096, 4096}}, buffer);
-    EXPECT_EQ(1, num_reserved);
-    EXPECT_NE(slice1, iovecs[0].mem_);
-    commitReservation(iovecs, num_reserved, buffer);
+    {
+      expectSlices({{1, 16383, 16384}}, buffer);
+      buffer.addBufferFragment(fragment);
+      EXPECT_EQ(13, buffer.length());
+      auto reservation = buffer.reserveApproximately(1);
+      EXPECT_EQ(1, reservation.numSlices());
+      EXPECT_NE(slice1, reservation.slices()[0].mem_);
+      reservation.commit(1);
+      expectSlices({{1, 16383, 16384}, {12, 0, 12}, {1, 16383, 16384}}, buffer);
+    }
     EXPECT_EQ(14, buffer.length());
   }
 }
 
 TEST_F(OwnedImplTest, ReserveCommitReuse) {
   Buffer::OwnedImpl buffer;
-  static constexpr uint64_t NumIovecs = 2;
-  Buffer::RawSlice iovecs[NumIovecs];
 
   // Reserve 8KB and commit all but a few bytes of it, to ensure that
   // the last slice of the buffer can hold part but not all of the
   // next reservation. Note that the buffer implementation might
   // allocate more than the requested 8KB. In case the implementation
   // uses a power-of-two allocator, the subsequent reservations all
   // request 16KB.
-  uint64_t num_reserved = buffer.reserve(8200, iovecs, NumIovecs);
-  EXPECT_EQ(1, num_reserved);
-  iovecs[0].len_ = 8000;
-  buffer.commit(iovecs, 1);
+  {
+    auto reservation = buffer.reserveSingleSlice(8200);
+    reservation.commit(8000);
+  }
   EXPECT_EQ(8000, buffer.length());
 
   // Reserve 16KB. The resulting reservation should span 2 slices.
   // Commit part of the first slice and none of the second slice.
-  num_reserved = buffer.reserve(16384, iovecs, NumIovecs);
-  EXPECT_EQ(2, num_reserved);
-  const void* first_slice = iovecs[0].mem_;
-  iovecs[0].len_ = 1;
-  expectSlices({{8000, 4288, 12288}, {0, 12288, 12288}}, buffer);
-  buffer.commit(iovecs, 1);
+  const void* first_slice;
+  {
+    auto reservation = buffer.reserveApproximately(16384);
+    expectSlices({{8000, 4288, 12288}}, buffer);
```

Comment:
Would it be possible to have a comment in some test indicating that it is expected no additional slices in the buffer after reserve since slices are added to the buffer on commit?